File: without-library/basics/sieve.cpp
#include <iostream>
#include <vector>

using namespace std;

// Manual Sieve
vector<int> sieve(int n) {
    vector<int> primes;
    vector<bool> is_prime(n + 1, true);
    is_prime[0] = is_prime[1] = false;
    
    for (int i = 2; i * i <= n; i++) {
        if (is_prime[i]) {
            for (int j = i * i; j <= n; j += i) {
                is_prime[j] = false;
            }
        }
    }
    
    for (int i = 2; i <= n; i++) {
        if (is_prime[i]) primes.push_back(i);
    }
    
    return primes;
}

int main() {
    return 0;
}

File: without-library/basics/sieve.py
import sys

# Manual Sieve of Eratosthenes

def sieve(n):
    primes = []
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False
    
    for i in range(2, int(n**0.5) + 1):
        if is_prime[i]:
            for j in range(i*i, n+1, i):
                is_prime[j] = False
                
    for i in range(2, n + 1):
        if is_prime[i]:
            primes.append(i)
            
    return primes

if __name__ == "__main__":
    pass
